Inventory.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace InventoryManagement
+{
+    public class Inventory
+    {
+
+        List<Product> products = new List<Product>();
+
+        //Define Constructor 
+        public Inventory() { }
+
+        //Add new Product to the List 
+
+        public void AddProduct(Product product)
+        {
+            products.Add(product);
+        }
+
+        //Delete Product from the List 
+
+        public void RemoveProduct(Product product)
+        {
+            products.Remove(product);
+        }
+
+        //Display All Products in the List 
+        public void ViewAllProducts()
+        {
+            foreach (var product in products)
+            {
+                Console.WriteLine("Name={0}\tQuantity={1}\tPrice={2}$", product.Name, product.Quantity, product.Price);
```

Comment:
1. Do you think that `Inventory` should be handling the console output logic?
2. Is it fine that this method _prints_ the products, while the `SearchProduct` _returns_ the product itself?

Reply:
1. Do you mean using the same strategy for printing, whether in the Inventory class or the main class?
2. The SearchProduct method is used in other functions.

Reply:
The `Inventory` class seems to be polluted with the printing logic, it should be responsible for handling products only. Outputing to the standard output doesn't sound like something that should be done in the `Inventory` class. Also, I think it would be better to have some sort of common interface for the class like if one method returns a `Product` object, I'd expect the other to return something similar (maybe a collection of `Product`).